app/assets/javascripts/components/contextual-guidance.js

```diff
@@ -0,0 +1,69 @@
+function ContextualGuidance ($module) {
+  if (!$module) $module = document
+  this.$module = $module
+  this.$fields = $module.querySelectorAll('[data-contextual-guidance]')
+}
+
+function nodeListForEach (nodes, callback) {
+  if (window.NodeList.prototype.forEach) {
+    return nodes.forEach(callback)
+  }
+  for (var i = 0; i < nodes.length; i++) {
+    callback.call(window, nodes[i], i, nodes)
+  }
+}
+
+ContextualGuidance.prototype.handleFocus = function (event) {
+  // Get the target element
+  var target = event.target
+  var guidanceId = target.getAttribute('data-contextual-guidance')
+
+  // If we have guidance for the field
+  if (guidanceId) {
+    var guidance = document.querySelector('#' + guidanceId)
+    ContextualGuidance.prototype.hideGuidance()
+    ContextualGuidance.prototype.showGuidance(guidance)
+  }
+}
+
+ContextualGuidance.prototype.showGuidance = function (element) {
+  if (element) {
+    element.classList.remove('govuk-visually-hidden')
+  }
+}
+
+ContextualGuidance.prototype.hideGuidance = function (element) {
```

Comment:
I can't see where this function is being called with a param - I think we always go into the 'else' clause. Do we need the param/if?